tests/core/Operation.spec.ts

```diff
@@ -34,4 +36,220 @@ describe('Operation', async () => {
+        } else {
+          throw error; // Unexpected error, throw to fail the test.
+        }
+      }
+    });
+
+    it('should throw error if `patches` is not an array.', async () => {
+      const updatePayload = generateUpdatePayloadForPublicKeys();
+      (updatePayload as any).patches = 'shouldNotBeAString';
+
+      const expectedError = new SidetreeError(ErrorCode.OperationUpdatePatchesNotArray);
+      expect(() => { Operation.validateUpdatePayload(updatePayload); }).toThrow(expectedError);
+    });
+
+    it('should throw error if given `action` is unknown.', async () => {
+      const updatePayload = generateUpdatePayloadForPublicKeys();
+      updatePayload.patches[0].action = 'invalidAction';
+
+      const expectedError = new SidetreeError(ErrorCode.OperationUpdatePatchMissingOrUnknownAction);
+      expect(() => { Operation.validateUpdatePayload(updatePayload); }).toThrow(expectedError);
+    });
+
+    it('should throw error if an add-public-keys patch contains additional unknown property.', async () => {
+      const updatePayload = generateUpdatePayloadForPublicKeys();
+      (updatePayload.patches[0] as any).unknownProperty = 'unknownProperty';
+
+      const expectedError = new SidetreeError(ErrorCode.OperationUpdatePatchMissingOrUnknownProperty);
+      expect(() => { Operation.validateUpdatePayload(updatePayload); }).toThrow(expectedError);
+    });
+
+    it('should throw error if `publicKeys` in an add-public-keys patch is not an array.', async () => {
+      const updatePayload = generateUpdatePayloadForPublicKeys();
+      (updatePayload.patches[0] as any).publicKeys = 'incorrectType';
+
+      const expectedError = new SidetreeError(ErrorCode.OperationUpdatePatchPublicKeysNotArray);
+      expect(() => { Operation.validateUpdatePayload(updatePayload); }).toThrow(expectedError);
+    });
+
+    it('should throw error if an entry in `publicKeys` in an add-public-keys patch contains additional unknown property.', async () => {
+      const updatePayload = generateUpdatePayloadForPublicKeys();
+      (updatePayload.patches[0].publicKeys![0] as any).unknownProperty = 'unknownProperty';
+
+      const expectedError = new SidetreeError(ErrorCode.OperationUpdatePatchPublicKeyMissingOrUnknownProperty);
+      expect(() => { Operation.validateUpdatePayload(updatePayload); }).toThrow(expectedError);
+    });
+
+    it('should throw error if `id` of a public key in an add-public-keys patch is not a string.', async () => {
+      const updatePayload = generateUpdatePayloadForPublicKeys();
+      (updatePayload.patches[0].publicKeys![0] as any).id = { invalidType: true };
+
+      const expectedError = new SidetreeError(ErrorCode.OperationUpdatePatchPublicKeyIdNotString);
+      expect(() => { Operation.validateUpdatePayload(updatePayload); }).toThrow(expectedError);
+    });
+
+    it('should throw error if the a secp256k1 public key in an add-public-keys patch is not in `publicKeyHex` format.', async () => {
```

Comment:
I know we are using compressed public key format, we should maybe add a test here to ensure that the none compressed public key format won't be added, because that could cause some strange errors later...

In particular, ethereum users are likely to try and register uncompressed keys since, they are used to calculate addresesses normally...

https://bitcoin.stackexchange.com/questions/44260/can-ethereum-support-compressed-secp256k1-public-keys-to-compute-keccak-addresse

Reply:
Will add a length check of 66 chars.